aggregator/src/main/scala/weco/concepts/aggregator/AggregatorMain.scala

```diff
@@ -0,0 +1,65 @@
+package weco.concepts.aggregator
+
+import com.typesafe.config.{Config, ConfigFactory}
+import net.ceedubs.ficus.Ficus._
+import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+import akka.actor.ActorSystem
+import grizzled.slf4j.Logging
+import weco.concepts.aggregator.secrets.{
+  ClusterConfWithSecrets,
+  SecretsResolver
+}
+import weco.concepts.aggregator.sources.WorkIdSource
+import weco.concepts.common.elasticsearch.Indexer
+
+import scala.concurrent.ExecutionContext
+
+/** Common base for the entrypoint to aggregator, regardless of how it is called
+  * This loads the appropriate configuration for the application, chosen by the
+  * AGGREGATOR_APP_CONTEXT environment variable. If absent, the application will
+  * run as though Elasticsearch is running on the same host.
+  *
+  * Implementations inheriting from this trait are expected to determine a
+  * source of Works JSON from their calling parameters and run the
+  * ConceptAggregator with it.
+  *
+  * One of the implementations that inherits from this trait is a Lambda
+  * function, Because of this, as much common setup as possible is done in the
+  * construction so that the only work that needs to be done on a per-call basis
+  * is that which differs per-call.
+  */
+trait AggregatorMain extends Logging {
```

Comment:
All of this has been moved from Main, so that Main and LambdaMain can have common construction WRT looking stuff up in config.